tests/phpunit/tests/post/getPageByTitle.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+/**
+ * @group post
+ * @covers ::get_page_by_title
+ */
+class Tests_Post_GetPageByTitle extends WP_UnitTestCase {
+	/**
+	 * @ticket 36905
+	 */
+	public function test_get_page_by_title_priority() {
+		$attachment = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'some-other-page',
+				'post_type'  => 'attachment',
+			)
+		);
+		$page       = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'some-page',
+				'post_type'  => 'page',
+			)
+		);
+
+		// get_page_by_title() should return a post of the requested type before returning an attachment.
+		$this->assertEquals( $page, get_page_by_title( 'some-page' ) );
+
+		// Make sure get_page_by_title() will still select an attachment when a post of the requested type doesn't exist.
+		$this->assertEquals( $attachment, get_page_by_title( 'some-other-page', OBJECT, 'attachment' ) );
+	}
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_should_match_top_level_page() {
+		$page = self::factory()->post->create(
+			array(
+				'post_type'  => 'page',
+				'post_title' => 'foo',
+			)
+		);
+
+		$found = get_page_by_title( 'foo' );
+
+		$this->assertSame( $page, $found->ID );
+	}
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_inherit() {
+		$page = self::factory()->post->create(
+			array(
+				'post_type'   => 'page',
+				'post_title'  => 'foo',
+				'post_status' => 'inherit',
+			)
+		);
+
+		$found = get_page_by_title( 'foo' );
+
+		$this->assertSame( $page, $found->ID );
+	}
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_should_obey_post_type() {
+		register_post_type( 'wptests_pt' );
+
+		$page = self::factory()->post->create(
+			array(
+				'post_type'  => 'wptests_pt',
+				'post_title' => 'foo',
+			)
+		);
+
+		$found = get_page_by_title( 'foo' );
+		$this->assertNull( $found );
+
+		$found = get_page_by_title( 'foo', OBJECT, 'wptests_pt' );
+		$this->assertSame( $page, $found->ID );
+	}
+
+
+	/**
+	 * @ticket 36905
+	 */
+	public function test_should_hit_cache() {
+		global $wpdb;
+
+		$page = self::factory()->post->create(
+			array(
+				'post_type'  => 'page',
+				'post_title' => 'foo',
+			)
+		);
+
+		// Prime cache.
+		$found = get_page_by_title( 'foo' );
+		$this->assertSame( $page, $found->ID );
+
+		$num_queries = $wpdb->num_queries;
```

Comment:
No need to use wpdb global, there's a function for that.

```suggestion
		$num_queries = get_num_queries();
```

:1234: This applies to all tests using the wpdb global.

Reply:
Fixed.